src/plots/cartesian/axes.js

```diff
@@ -1684,7 +1684,8 @@ axes.doTicks = function(gd, axid, skipTitle) {
                     fontSize*(offsetBase + (ax.showticklabels ? 1.5 : 0.5)));
 
             if(ax.rangeslider && ax.rangeslider.visible && ax._boundingBox) {
-                y += gs.h * ax.rangeslider.thickness + ax._boundingBox.height;
+                y += (fullLayout.height - fullLayout.margin.b - fullLayout.margin.t) *
+                    ax.rangeslider.thickness + ax._boundingBox.height;
```

Comment:
Yep. I wasn't sure on the best way to deal with finding the space needed for positioning the rangeslider - if you have any ideas let me know. Making it work consistently with svg and gl is a bit of a challenge too (from my still-exploring-the-codebase point of view).